### PR TITLE
add https/http proxy support

### PR DESCRIPTION
**Pull request by vdeturckheim**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,15 @@ const Binding = require(binding_path);
 (You are, of courses, free to get the `binary` part of the `package.json` by any
 other way that works for you!);
 
+## Proxy support
+
+This module supports https proxy. If an URL is present in the `https_proxy`, the 
+`HTTPS_PROXY` or the `npm_config_proxy` environment variable, the download
+of the binary will be done thorough this URL using the `https-proxy-agent`
```
Comment (mauricesvay) [inline README.md]:
```suggestion
of the binary will be done through this URL using the `https-proxy-agent`
```

**Review by rumpl — APPROVED** [review]
LGTM 👍

File: package.json [hunk 2]
```diff
@@ -24,6 +24,8 @@
   "homepage": "https://github.com/vdeturckheim/binary-downloader#readme",
   "dependencies": {
     "detect-libc": "^1.0.3",
+    "http-proxy-agent": "^4.0.1",
+    "https-proxy-agent": "^5.0.0",
```
Comment (YoannMa) [inline package.json:28]: `node-http-proxy-agent@v4.0.1` require node 6+
`node-https-proxy-agent@v5.0.0` require node 6+

**Review by YoannMa — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (vdeturckheim) [reply package.json:28]: T_T

**Pull request merged**